### PR TITLE
1 develop auto testing suit for monorepo

### PR DESCRIPTION
**Pull request by kefeimo**
### Sumary

* Created pytest suits for both sub-mono-repos.
    * Including .github/workflows.
* Pre-run workflows result
    * [1 develop auto testing suit for monorepo](https://github.com/kefeimo/volttron-boptest/actions/runs/5732902646) (for Run volttron-boptest-agent Pytests)
    * [1 develop auto testing suit for monorepo](https://github.com/kefeimo/volttron-boptest/actions/runs/5732902637) (for Run volttron-lib-boptest-integration Pytests)

### CODE REVIEW

File: .github/workflows/run-volttron-boptest-agent-tests.yml [hunk 1]
```diff
@@ -60,10 +60,53 @@ jobs:
...
           installer-parallel: true
 
+      #----------------------------------------------
+      # Setup boptest server at local
+      #----------------------------------------------
+      - name: Git clone Boptest Repo
+        run:
+          git clone https://github.com/ibpsa/project1-boptest.git
+
+#      - name: validate repo
```
Comment (craigpnnl) [inline .github/workflows/run-volttron-boptest-agent-tests.yml]: please remove the commented out code unless there for documentation of purpose.

File: .github/workflows/run-volttron-lib-boptest-integration-tests.yml [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+#      - name: validate repo
+#        run:
+#          ls -la project1-boptest
+
+      - name: install docker-compose
+        run: |
+          sudo curl -L https://github.com/docker/compose/releases/download/1.29.2/docker-compose-`uname -s`-`uname -m` -o /usr/local/bin/docker-compose
+          sudo chmod +x /usr/local/bin/docker-compose
+#      - name: install docker-compose
```
Comment (craigpnnl) [inline .github/workflows/run-volttron-lib-boptest-integration-tests.yml]: remove commented out code that aren't relevent

**Review by craigpnnl — CHANGES_REQUESTED** [review]

File: volttron-boptest-agent/tests/test_boptest_agent.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+import logging
+
+logging_logger = logging.getLogger(__name__)
+logging_logger.setLevel(logging.INFO)
+
+boptest_vip_identity = "volttron_boptest_agent"
+
+
+@pytest.fixture(scope="module")
```
Comment (craigpnnl) [inline volttron-boptest-agent/tests/test_boptest_agent.py]: I don't understand why you are doing this.

The volttron instance creates a new volttron_home for every test instance.  Why are we testing this here?

File: volttron-boptest-agent/tests/test_boptest_agent.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+def test_volttron_instance_fixture(volttron_instance):
+    print(volttron_instance)
+    logging_logger.info(f"=========== volttron_instance_new.volttron_home: {volttron_instance.volttron_home}")
+    logging_logger.info(f"=========== volttron_instance_new.skip_cleanup: {volttron_instance.skip_cleanup}")
+    logging_logger.info(f"=========== volttron_instance_new.vip_address: {volttron_instance.vip_address}")
+
+
+@pytest.fixture(scope="module")
+def vip_agent(volttron_instance):
```
Comment (craigpnnl) [inline volttron-boptest-agent/tests/test_boptest_agent.py]: These tests are already done in the volttron-testing library

File: volttron-lib-boptest-integration/tests/conftest.py [hunk 5]
```diff
@@ -22,7 +22,7 @@
 # ===----------------------------------------------------------------------===
 # }}}
 
-"""Configuration for the pytest test suite."""
+"""Configuration for the pytest test suits (make sure /src is resolvable)."""
```
Comment (craigpnnl) [inline volttron-lib-boptest-integration/tests/conftest.py]: should be suite not suit

File: volttron-boptest-agent/tests/test_boptest_agent.py [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+    pid = volttron_instance.start_agent(uuid)
+    gevent.sleep(5)
+    logging_logger.info(
+        f"=========== volttron_instance.is_agent_running(uuid): {volttron_instance.is_agent_running(uuid)}")
+    # TODO: get retry_call back
+    return {"uuid": uuid, "pid": pid}
+
+
+def test_install_boptest_agent_fixture(boptest_agent, vip_agent, volttron_instance):
+    puid = boptest_agent
```
Comment (craigpnnl) [inline volttron-boptest-agent/tests/test_boptest_agent.py:78]: this isn't necessary...you can use the dynamic_agent that is started for each instance to do peerlist on.

**Review by craigpnnl — CHANGES_REQUESTED** [review]

File: volttron-boptest-agent/tests/test_boptest_agent.py (hunk 3 above)
Comment (kefeimo) [reply volttron-boptest-agent/tests/test_boptest_agent.py]: This is mainly for debugging purpose, i.e., to display what volttron home is. The intention is to make sure each test unit is working and help to narrow down the root cause. 
Should I
* skip the test
* comment out the test
* or, delete the code block?

File: volttron-boptest-agent/tests/test_boptest_agent.py (hunk 4 above)
Comment (kefeimo) [reply volttron-boptest-agent/tests/test_boptest_agent.py]: The intention is to make sure each test unit is working and help to narrow down the root cause. 
Should I
* skip the test
* comment out the test
* or, delete the code block?

File: volttron-lib-boptest-integration/tests/conftest.py (hunk 5 above)
Comment (kefeimo) [reply volttron-lib-boptest-integration/tests/conftest.py]: resolved

File: volttron-boptest-agent/tests/test_boptest_agent.py (hunk 6 above)
Comment (kefeimo) [reply volttron-boptest-agent/tests/test_boptest_agent.py:78]: I am following the pattern of volttron-lib-fake-driver: https://github.com/eclipse-volttron/volttron-lib-fake-driver/blob/main/tests/test_driver.py#L90.

Is there an example of using dynamic_agent? Does it have Python api or we will utilize subprocess to call from command line?

File: .github/workflows/run-volttron-boptest-agent-tests.yml (hunk 1 above)
Comment (kefeimo) [reply .github/workflows/run-volttron-boptest-agent-tests.yml]: resolved

File: .github/workflows/run-volttron-lib-boptest-integration-tests.yml (hunk 2 above)
Comment (kefeimo) [reply .github/workflows/run-volttron-lib-boptest-integration-tests.yml]: resolved

**Review by kefeimo — COMMENTED** [review]
changed based @craig8 's feedback

**Comment by kefeimo** [comment]
@craig8, thanks for the review. I responded inline. (Somehow github doesn't give me the option to re-request review.)

**Review by craigpnnl — APPROVED** [review]

**Review by shwethanidd — APPROVED** [review]

**Pull request merged**